src/Motor.Extensions.Hosting.RabbitMQ/RabbitMQMessageConsumer.cs

```diff
@@ -89,87 +90,115 @@ private void ThrowIfConsumerAlreadyStarted()
         }
     }
 
-    private void ConfigureChannel()
+    private async Task ConfigureChannelAsync(CancellationToken ct)
     {
-        _channel?.BasicQos(0, _options.PrefetchCount, false);
+        if (_channel != null)
+        {
+            await _channel.BasicQosAsync(0, _options.PrefetchCount, false, ct);
+        }
     }
 
-    private void DeclareQueue()
+    private async Task DeclareQueueAsync()
     {
         if (!_options.DeclareQueue)
         {
             return;
         }
 
-        DeclareAndBindConsumerQueue();
-        DeclareAndBindConsumerDeadLetterExchangeQueue();
+        await DeclareAndBindConsumerQueueAsync();
+        await DeclareAndBindConsumerDeadLetterExchangeQueueAsync();
     }
 
-    private void StartConsumerOnChannel()
+    private async Task StartConsumerOnChannelAsync()
     {
-        var consumer = new EventingBasicConsumer(_channel);
-        consumer.Received += (_, args) => ConsumerCallback(args);
+        if (_channel is null)
+        {
+            _logger.LogError(LogEvents.ConsumerNotStarted,
+                "Message consumer could not start because channel is null");
+            _applicationLifetime.StopApplication();
+            return;
+        }
+        var consumer = new AsyncEventingBasicConsumer(_channel);
+        consumer.ReceivedAsync += (_, args) =>
+        {
+#pragma warning disable CS4014
```

Comment:
(nitpick) I think a comment explaining why we explicitly **do not** await here would be good to have.
Otherwise we will wonder about this again in the future and might want to add the `await` again.